osu.Game/Overlays/DashboardOverlay.cs

```diff
@@ -0,0 +1,179 @@
+
+        public DashboardOverlay()
+            : base(OverlayColourScheme.Purple)
+        {
+            Children = new Drawable[]
+            {
+                background = new Box
+                {
+                    RelativeSizeAxes = Axes.Both
+                },
+                scrollFlow = new BasicScrollContainer
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    ScrollbarVisible = false,
+                    Child = new FillFlowContainer
+                    {
+                        AutoSizeAxes = Axes.Y,
+                        RelativeSizeAxes = Axes.X,
+                        Direction = FillDirection.Vertical,
+                        Children = new Drawable[]
+                        {
+                            header = new DashboardOverlayHeader
+                            {
+                                Anchor = Anchor.TopCentre,
+                                Origin = Anchor.TopCentre,
+                                Depth = -float.MaxValue
+                            },
+                            content = new Container
+                            {
+                                RelativeSizeAxes = Axes.X,
+                                AutoSizeAxes = Axes.Y
+                            }
+                        }
+                    }
+                },
+                loading = new LoadingLayer(content),
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            background.Colour = ColourProvider.Background5;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            localUser.BindTo(api.LocalUser);
+            localUser.BindValueChanged(_ => onTabChanged());
+            header.Current.BindValueChanged(_ => onTabChanged());
+        }
+
+        protected override void PopIn()
+        {
+            base.PopIn();
+
+            if (!content.Any())
+                header.Current.TriggerChange();
+        }
+
+        protected override void PopOutComplete()
+        {
+            base.PopOutComplete();
+            loadDisplay(null);
+        }
+
+        private void onTabChanged()
+        {
+            lastRequest?.Cancel();
+            cancellationToken?.Cancel();
+
+            loading.Show();
+
+            // We may want to use OnlineViewContainer after https://github.com/ppy/osu/pull/8044 merge
+            if (!api.IsLoggedIn)
+            {
+                loadDisplay(null);
+                return;
+            }
+
+            var request = createScopedRequest();
+            lastRequest = request;
+
+            if (request == null)
+            {
+                loadDisplay(null);
+                return;
+            }
+
+            request.Success += () => Schedule(() => loadDisplay(createDisplayFromResponse(request)));
+            request.Failure += _ => Schedule(() => loadDisplay(null));
```

Comment:
```suggestion
            request.Failure += _ => Schedule(() => loadDisplay(Empty()));
```
although all things considered this probably isn't the best UX (but I suppose this can be addressed in a separate PR further down the line).